packages/common/src/index.ts

```diff
@@ -230,11 +230,18 @@ export enum DimensionType {
     BOOLEAN = 'boolean',
 }
 
+export enum TimeInterval {
+    DAY = 'DAY',
+    WEEK = 'WEEK',
+    MONTH = 'MONTH',
+    YEAR = 'YEAR',
+}
+
 export interface Dimension extends Field {
     fieldType: FieldType.DIMENSION;
     type: DimensionType;
     group?: string;
-    timeInterval?: string;
+    timeInterval?: TimeInterval | string;
```

Comment:
Why can this accept `string` ?

Reply:
So it could work with whatever time intervals their warehouse support and be case insensitive. E.g some warehouse support 'QUARTER'  but others don't. 
In that case the FE would default to the generic date picker.